src/visitors/units_visitor.cpp

```diff
@@ -146,7 +147,7 @@ void UnitsVisitor::visit_factor_def(ast::FactorDef& node) {
         auto node_unit_name = node.get_node_name();
         auto unit1_factor = units_driver.table->get_unit(node_unit_name + "_unit1")->get_factor();
         auto unit2_factor = units_driver.table->get_unit(node_unit_name + "_unit2")->get_factor();
-        auto unit_factor = unit1_factor / unit2_factor;
+        auto unit_factor = stringutils::to_string(unit1_factor / unit2_factor);
```

Comment:
Here we should take care of the precision of the double converted to float according to the legacy or non legacy units similar to:
https://github.com/BlueBrain/nmodl/pull/421/files#diff-ecbcbc7e327becd3a246884e25ddd4d143b61bc35a7e39e68e4d5bf69d9b84eaR1997
Also the above should be edited to print a string